packages/frontend/src/ui/popover/index.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+    forwardRef,
+    type ReactElement,
+    type ReactNode,
+    useState,
+    useEffect,
+} from "react";
+import { usePopper } from "react-popper";
+import { type Placement } from "@popperjs/core";
+
+import styles from "./styles.module.css";
+
+export interface PopoverProps {
+    open: boolean;
+    anchor?: Element | null;
+    placement?: Placement;
+    offset?: [number, number];
+    className?: { root?: string };
+    children?: ReactNode;
+}
+
+export const Popover = forwardRef<HTMLDivElement, PopoverProps>(
+    function Popover(
+        {
+            open,
+            anchor,
+            placement = "auto",
+            offset = [0, 8],
+            className,
+            children,
+        },
+        ref,
+    ): ReactElement {
+        const [popper, setPopper] = useState<HTMLDivElement | null>(null);
+
+        const {
+            styles: popperStyles,
+            attributes,
+            update,
+        } = usePopper(anchor, popper, {
+            placement,
+            modifiers: [
+                {
+                    name: "offset",
+                    options: { offset },
+                },
+            ],
+        });
+
+        useEffect(() => {
+            if (open && update) update();
+        }, [open, update]);
+
+        return (
+            <div
+                ref={(element) => {
+                    if (ref) {
+                        if (typeof ref === "function") ref(element);
+                        else ref.current = element;
+                    }
+                    setPopper(element);
+                }}
+                style={{ ...popperStyles.popper }}
+                className={`${styles.root} ${className} ${open ? styles.open : ""}`}
```

Comment:
Here is should be `className.root` no?

Reply:
Nono `className` is right, but I needed to change the prop type to `string`.

Reply:
Addressed in 72a15a7b69529c090da6887d86c5a5f2d3fb8b08